module-2/lab-matplotlib-seaborn/your-code/challenge-2.ipynb

```diff
@@ -23,18 +23,16 @@
   },
```

Comment:
- Please apply the 2 principles : KISS and less is more. The following chunk of code is not necessary: 
```Python 
v_2016 = fitbit_mounth.loc[0:3]
v_2015 = fitbit_mounth.loc[4:]
pd.concat([v_2015,v_2016])
```
You can solve the 2 questions here with the following:
```Python
# Median Steps by Month_encoded
weekday_steps = fitbit['Steps'].groupby(fitbit['Months_encoded']).median()
weekday_steps.head(5)

fig,axes = plt.subplots(figsize = (12,4),nrows = 1, ncols = 2)
plt.sca(axes[0])
weekday_steps.plot(kind = 'line',linewidth=2)
plt.ylabel('Median number of steps')
plt.xlabel('Months')
plt.title('Median Steps walked by month')
plt.xticks(list(range(13)),['','Jan','Feb','Mar','Apr','May','Jun','Jul','Aug','Sep','Oct','Nov','Dec'])

plt.savefig('steps_months.png',bbox_inches='tight')


# Median Steps by Work_or_Weekend
plt.sca(axes[1])
weekday_steps = fitbit['Steps'].groupby(fitbit['Work_or_Weekend']).median().sort_values()
weekday_steps.plot(kind = 'bar',alpha = 0.5)
plt.ylabel('Median number of steps')
plt.title('Median Steps walked by Workdays and Weekend')
plt.xticks(list(range(2)),['Weekend','Workdays'])
```